crates/bevy_render/src/camera/camera.rs

```diff
@@ -373,6 +373,39 @@ impl Camera {
         Some(viewport_position)
     }
 
+    /// Given a position in world space, use the camera to compute the viewport-space coordinates and depth.
+    ///
+    /// To get the coordinates in Normalized Device Coordinates, you should use
+    /// [`world_to_ndc`](Self::world_to_ndc).
+    ///
+    /// Returns `None` if any of these conditions occur:
```

Comment:
Non-blocking: this feels like a case where a `Result` would be clearer / more helpful for debugging: multiple failure modes feel like a custom error type is the right choice.

Reply:
Same here: just an extended version of `world_to_viewport`.
Could do the same there, but maybe in a follow up PR?
That would also make it a breaking change.

Reply:
Yep, let's swap all of these over in a follow-up.